### PR TITLE
Random fixes

### PR DESCRIPTION
**Pull request by paplorinc**
A few insignificant changes that I noticed while browsing around the code.

### CODE REVIEW

File: javaslang/src/main/java/javaslang/collection/Array.java [hunk 1]
```diff
@@ -26,10 +26,10 @@
 
     private static final Array<?> EMPTY = new Array<>(new Object[0]);
 
-    private final Object[] back;
+    private final Object[] delegate;
```
Comment (paplorinc) [inline javaslang/src/main/java/javaslang/collection/Array.java]: All the operations are delegated to this entity, hence the rename.
`back` can have too many meanings.

File: javaslang/src/main/java/javaslang/collection/Array.java [hunk 2]
```diff
@@ -760,45 +760,41 @@ private Object readResolve() {
...
+        System.arraycopy(delegate, index, arr, index + 1, delegate.length - index);
         return wrap(arr);
     }
 
     @Override
     public Array<T> insertAll(int index, Iterable<? extends T> elements) {
-        if (index < 0) {
-            throw new IndexOutOfBoundsException("insert(" + index + ", e)");
-        } else if (index > length()) {
+        if (index < 0 || index > length()) {
```
Comment (paplorinc) [inline javaslang/src/main/java/javaslang/collection/Array.java]: merged

File: javaslang/src/main/java/javaslang/collection/Collections.java [hunk 3]
```diff
@@ -101,10 +106,10 @@ static int hash(Iterable<?> iterable) {
         return hashCode;
     }
 
-    static <T, U, C extends Iterable<U>, R extends Traversable<U>> R scanLeft(
-            Iterable<? extends T> elements,
-            U zero, BiFunction<? super U, ? super T, ? extends U> operation,
-            C cumulativeResult, BiFunction<C, U, C> combiner, Function<C, R> finisher) {
+    static <T, U, C extends Iterable<U>, R extends Traversable<U>> R scanLeft(Iterable<? extends T> elements,
```
Comment (paplorinc) [inline javaslang/src/main/java/javaslang/collection/Collections.java]: minor: had inconsistent formatting

File: javaslang/src/main/java/javaslang/collection/Collections.java [hunk 4]
```diff
@@ -183,7 +187,7 @@ protected T getNext() {
     }
 
     @SuppressWarnings("unchecked")
-    private static <T> Seq<T> seq(Iterable<? extends T> iterable) {
+    static <T> Seq<T> seq(Iterable<? extends T> iterable) {
```
Comment (paplorinc) [inline javaslang/src/main/java/javaslang/collection/Collections.java]: this was needed for the new `Vector` impl

File: javaslang/src/main/java/javaslang/collection/List.java [hunk 5]
```diff
@@ -737,17 +737,13 @@
...
         Objects.requireNonNull(mapper, "mapper is null");
-        if (isEmpty()) {
-            return empty();
-        } else {
-            List<U> list = empty();
-            for (T t : this) {
-                for (U u : mapper.apply(t)) {
-                    list = list.prepend(u);
-                }
+        List<U> list = empty();
```
Comment (paplorinc) [inline javaslang/src/main/java/javaslang/collection/List.java]: no need for the if

File: javaslang/src/main/java/javaslang/collection/Collections.java [hunk 6]
```diff
@@ -114,11 +119,11 @@ static int hash(Iterable<?> iterable) {
         return finisher.apply(cumulativeResult);
     }
 
-    static <T, U, C extends Iterable<U>, R extends Traversable<U>> R scanRight(
-            Iterable<? extends T> elements,
-            U zero, BiFunction<? super T, ? super U, ? extends U> operation,
-            C cumulativeResult, BiFunction<C, U, C> combiner, Function<C, R> finisher) {
-        final Iterator<? extends T> reversedElements = seq(elements).reverseIterator();
+    static <T, U, C extends Iterable<U>, R extends Traversable<U>> R scanRight(Iterable<? extends T> elements,
+                                                                               U zero, BiFunction<? super T, ? super U, ? extends U> operation,
```
Comment (danieldietrich) [inline javaslang/src/main/java/javaslang/collection/Collections.java]: which formatter do you use that inserts that much spaces?? I've recognized that before

File: javaslang/src/main/java/javaslang/collection/Collections.java [hunk 7]
```diff
@@ -114,11 +119,11 @@ static int hash(Iterable<?> iterable) {
...
+                                                                               C cumulativeResult, BiFunction<C, U, C> combiner,
+                                                                               Function<C, R> finisher) {
+        final Iterator<? extends T> reversedElements = seq(elements).reverseIterator(); // TODO a List will be reversed too many times this way
```
Comment (danieldietrich) [inline javaslang/src/main/java/javaslang/collection/Collections.java]: We should add no TODOs in code (I know, I've also done it before). Better create an issue.

**Comment by danieldietrich** [comment]
Thank you!

**Pull request merged**